src/backend/utilities/trackConnectionState.ts

```diff
@@ -0,0 +1,20 @@
+export function trackConnectionState(toleratedOffDurationMs: number) {
+  let offSince: Date | undefined = new Date();
+
+  return {
+    on() {
+      offSince = undefined;
+    },
+    off() {
+      offSince = new Date();
+    },
+    offForTooLong() {
```

Comment:
for functions with boolean return type, I would prefer an `is` prefix. `isOffForTooLong` in this case

Reply:
We do not apply it as a general rule, but here it makes sense